presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -896,7 +899,12 @@ protected String quoted(String name)
         return identifierQuote + name + identifierQuote;
     }
 
-    protected String quoted(String catalog, String schema, String table)
+    protected String quoted(RemoteTable remoteTable)
+    {
+        return quoted(remoteTable.getCatalogName().orElse(null), remoteTable.getSchemaName().orElse(null), remoteTable.getTableName());
+    }
+
+    protected String quoted(@Nullable String catalog, @Nullable String schema, String table)
```

Comment:
Maybe add TODO in the body, to remove special treatment for `""` values.
(`isNullOrEmpty` => `!= null`)

Reply:
I am not sure. That would require me to know that special value `""` is not special in some database. Maybe there a RDBMS that uses empty string instead of `null`?

Reply:
I agree